localstack-core/localstack/services/dynamodb/server.py

```diff
@@ -79,6 +79,10 @@ def get() -> "DynamodbServer":
     def start_dynamodb(self) -> bool:
         """Start the DynamoDB server."""
 
+        # We want this method to be idempotent.
+        if self.is_running() and self.is_up():
+            return True
```

Comment:
Would this be invoked in a multi-threaded environment? If so, this should happen within the context of a mutex.

Reply:
Good point! I annotated the `start_dynamodb` function as we already do with `stop_dynamodb`. I think this should suffice.